crates/pallet-domains/src/domain_registry.rs

```diff
@@ -368,16 +368,18 @@ pub(crate) fn do_update_domain_allow_list<T: Config>(
     domain_id: DomainId,
     updated_operator_allow_list: OperatorAllowList<T::AccountId>,
 ) -> Result<(), Error> {
-    DomainRegistry::<T>::try_mutate(domain_id, |maybe_domain_object| {
-        let domain_obj = maybe_domain_object.as_mut().ok_or(Error::DomainNotFound)?;
-        ensure!(
-            domain_obj.owner_account_id == domain_owner,
-            Error::NotDomainOwner
-        );
+    let mut domain_obj =
+        Pallet::<T>::domain_registry_fallback(domain_id).ok_or(Error::DomainNotFound)?;
```

Comment:
The issue only shortly exists in runtime API calls on the block that contains the runtime upgrade.

While `do_update_domain_allow_list` (and also `do_register_operator`) is only called during block execution, where it is either using the old runtime and before migration or the new runtime and after the migration, it should not have the issue so not needed to use `domain_registry_fallback`.

Though it is also no harm to use `domain_registry_fallback` thus non-blocker.

Reply:
fine by me 👍🏼